### PR TITLE
tidb-lightning: update docs

### PR DESCRIPTION
**Pull request by Joyinqin**
### What is changed, added or deleted? (Required)
Translate changes of updated docs.

- TLS configuration
- TiKV Importer now supports listening on 0.0.0.0
- TiKV Importer no longer requires Pushgateway for Prometheus
- CSV strict-format config for optional parallel speed up
- TiDB's default SQL mode is no longer strict
- Several new CLI parameters are exposed

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/2946
- Other reference link(s):

### CODE REVIEW

**Comment by yikeke** [comment]
Please help review this PR, thanks! @lilin90

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 1]
```diff
@@ -136,6 +137,24 @@ TiDB Lightning 并不完全支持 `LOAD DATA` 语句中的所有配置项。例
 * 不可跳过表头（`IGNORE n LINES`）。如有表头，必须是有效的列名。
 * 定界符和分隔符只能为单个 ASCII 字符。
 
+## `strict-format`
+
+TiDB Lightning 在输入文件大小统一约为 `256 MB` 时，使用效果最佳。 如果输入单个巨大的 CSV 文件，TiDB Lightning 就只能使用一个线程来处理，这会极大降低导入速度。
```
Comment (kennytm) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:142]:
```suggestion
TiDB Lightning 在输入文件大小统一约为 `256 MB` 时，使用效果最佳。如果输入单个巨大的 CSV 文件，TiDB Lightning 就只能使用一个线程来处理，这会极大降低导入速度。
```

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 2]
```diff
@@ -136,6 +137,24 @@ TiDB Lightning 并不完全支持 `LOAD DATA` 语句中的所有配置项。例
...
+
+想要解决此问题，建议先将 CSV 文件拆分为多个文件。 对于通用 CSV 格式，在没有读取整个文件的情况下无法快速确定行的开始和结束时间。 因此 TiDB Lightning 默认不会自动拆分 CSV 文件。但如果确定输入的 CSV 文件符合限制要求，则可以启用 `strict-format` 设置，TiDB Lightning 则会将大文件拆分为多个 `256 MB` 文件进行并行处理。
```
Comment (kennytm) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:144]:
```suggestion
想要解决此问题，建议先将 CSV 文件拆分为多个文件。对于通用 CSV 格式，在没有读取整个文件的情况下无法快速确定行的开始和结束位置。因此 TiDB Lightning 默认不会自动拆分 CSV 文件。但如果确定输入的 CSV 文件符合限制要求，则可以启用 `strict-format` 设置，TiDB Lightning 则会将大文件拆分为多个 `256 MB` 文件进行并行处理。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 3]
```diff
@@ -132,6 +141,16 @@ no-schema = false
 # 注意：**数据** 文件始终解析为 binary 文件。
 character-set = "auto"
 
+# 假设输入数据格式严格以加快处理速度。
+# strict-format = true：
+#  * 在 CSV 文件中，即使使用引号，每个值也不能包含字面值换行符（U + 000A 和 U + 000D 或 \r 和 \n），即严格将换行符用于分隔行。
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-configuration.md:146]:
```suggestion
#  * 在 CSV 文件中，即使使用引号，每个值也不能包含字面值换行符（U+000A 和 U+000D, 即 \r 和 \n），即严格将换行符用于分隔行。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 4]
```diff
@@ -132,6 +141,16 @@ no-schema = false
...
+# 如果输入数据格式严格，TiDB Lightning 则会快速定位大文件的拆分位置进行并行处理。 但是如果输入数据为非严格格式，则可能会将有效数据分成两半，从而破坏结果。
+# 默认值为 false 时是安全处理速度。
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-configuration.md:148]: what does this mean

File: tidb-lightning/tidb-lightning-configuration.md [hunk 5]
```diff
@@ -217,13 +253,18 @@ log-file = "tikv-importer.log"
 # 日志等级：trace, debug, info, warn, error 和 off
 log-level = "info"
 
+# TiKV Importer 服务器的监听地址。 
+Prometheus 可以从这个地址抓取指标。
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-configuration.md:257]:
```suggestion
# Prometheus 可以从这个地址抓取指标。
```

**Comment by kennytm** [comment]
Why do we quote "256 MB" as ``` `256 MB` ``` now? File size is not code.

File: tidb-lightning/tidb-lightning-configuration.md (hunk 4 above)
Comment (Joyinqin) [reply tidb-lightning/tidb-lightning-configuration.md]: I have modified this paragraph. Is it right? PTAL @kennytm

File: tidb-lightning/tidb-lightning-configuration.md [hunk 6]
```diff
@@ -132,6 +141,19 @@ no-schema = false
 # 注意：**数据** 文件始终解析为 binary 文件。
 character-set = "auto"
 
+# 如果输入数据源为严格格式，则会加快处理速度。
+# 当 strict-format = true：
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-configuration.md:145]:
```suggestion
# strict-format = true 要求：
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 7]
```diff
@@ -176,12 +198,29 @@ index-serial-scan-concurrency = 20
...
 max-allowed-packet = 67_108_864
 
+# 是否将 TLS 用于 SQL 连接。当有效值为：
+#  * ""            - 如果填充了[tidb.security]部分，则强制使用 TLS（与 "cluster" 情况相同），否则与 "false" 情况相同
+#  * "false"       - 禁用 TLS
+#  * "cluster"     - 强制使用 TLS 并使用 [tidb.security] 部分中指定的 CA 验证服务器的证书
+#  * "skip-verify" - 强制使用 TLS，但不验证服务器的证书（不安全！）
+#  * "preferred"   - 与 "skip-verify" 相同，但是如果服务器不支持 TLS，则会退回到未加密的连接
+# tls = ""
+# 指定启用 TLS 的 MySQL 连接的证书和密钥。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:214]:
```suggestion
# 指定证书和密钥用于 TLS 连接 MySQL。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 8]
```diff
@@ -176,12 +198,29 @@ index-serial-scan-concurrency = 20
...
+# 默认为[security]部分的副本。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:215]:
```suggestion
# 默认为 [security] 部分的副本。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 9]
```diff
@@ -176,12 +198,29 @@ index-serial-scan-concurrency = 20
...
+# [tidb.security]
+# CA 的公共证书。 设置为空字符串可禁用 SQL 的 TLS。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:217]:
```suggestion
# CA 的公钥证书。设置为空字符串可禁用 SQL 的 TLS。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 10]
```diff
@@ -176,12 +198,29 @@ index-serial-scan-concurrency = 20
...
+# ca-path = "/path/to/ca.pem"
+# 此服务的公共证书。 默认为 `security.cert-path` 的副本
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:219]:
```suggestion
# 该服务的公钥证书。默认为 `security.cert-path` 的副本
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 11]
```diff
@@ -176,12 +198,29 @@ index-serial-scan-concurrency = 20
...
+# cert-path = "/path/to/lightning.pem"
+# 此服务的私钥。 默认为 `security.key-path` 的副本
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:221]:
```suggestion
# 此服务的私钥。默认为 `security.key-path` 的副本
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 12]
```diff
@@ -217,13 +256,18 @@ log-file = "tikv-importer.log"
 # 日志等级：trace, debug, info, warn, error 和 off
 log-level = "info"
 
+# TiKV Importer 服务器的监听地址。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:259]:
```suggestion
# 状态服务器的监听地址。 
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 13]
```diff
@@ -217,13 +256,18 @@ log-file = "tikv-importer.log"
 # 日志等级：trace, debug, info, warn, error 和 off
 log-level = "info"
 
+# TiKV Importer 服务器的监听地址。 
+# Prometheus 可以从这个地址抓取指标。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:260]:
```suggestion
# Prometheus 可以从这个地址抓取监控指标。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 14]
```diff
@@ -217,13 +256,18 @@ log-file = "tikv-importer.log"
...
+
 [server]
 # tikv-importer 的监听地址，tidb-lightning 需要连到这个地址进行数据写入。
-addr = "192.168.20.10:8287"
+addr = "0.0.0.0:8287"
 # gRPC 服务器的线程池大小。
 grpc-concurrency = 16
 
 [metric]
+# 当使用 Prometheus Pushgateway 时会涉及相关设置。通常可以通过 Prometheus 从 TikV Importer 服务器地址中抓取指标。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:270]:
```suggestion
# 当使用 Prometheus Pushgateway 时会涉及相关设置。通常可以通过 Prometheus 从 状态服务器地址中抓取指标。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 15]
```diff
@@ -298,6 +348,15 @@ min-available-ratio = 0.05
 | --tidb-status *port* | TiDB Server 的状态端口的（默认为 10080） | `tidb.status-port` |
 | --tidb-user *user* | 连接到 TiDB 的用户名 | `tidb.user` |
 | --tidb-password *password* | 连接到 TiDB 的密码 | `tidb.password` |
+| --no-schema | 先忽略表结构文件，直接从 TiDB 中获取表结构信息 | `mydumper.no-schema` |
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:351]:
```suggestion
| --no-schema | 忽略表结构文件，直接从 TiDB 中获取表结构信息 | `mydumper.no-schema` |
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 16]
```diff
@@ -298,6 +348,15 @@ min-available-ratio = 0.05
...
+| --enable-checkpoint *bool* | 是否启用断点 (默认值为 true) | `checkpoint.enable` |
+| --analyze *bool* | 导入后统计信息分析 (默认值为 true) | `post-restore.analyze` |
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:353]:
```suggestion
| --analyze *bool* | 导入后分析表信息 (默认值为 true) | `post-restore.analyze` |
```

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 17]
```diff
@@ -136,6 +137,24 @@ TiDB Lightning 并不完全支持 `LOAD DATA` 语句中的所有配置项。例
 * 不可跳过表头（`IGNORE n LINES`）。如有表头，必须是有效的列名。
 * 定界符和分隔符只能为单个 ASCII 字符。
 
+## `strict-format`
```
[the remark this answers is not in the document]
Comment (Joyinqin) [reply tidb-lightning/migrate-from-csv-using-tidb-lightning.md:140]: 是否会引起误解？ 比如 strict mode

File: tidb-lightning/deploy-tidb-lightning.md [hunk 18]
```diff
@@ -197,17 +197,12 @@ TiDB Lightning 可随 TiDB 集群一起用 [TiDB Ansible 部署](/online-deploym
     # 日志等级：trace、debug、info、warn、error、off。
     log-level = "info"
 
+    # TiKV Importer 服务器的监听地址。
```
[the remark this answers is not in the document]
Comment (Joyinqin) [reply tidb-lightning/deploy-tidb-lightning.md:200]: 这里不用明确说明吗

[the remark this answers is not in the document]
Comment (TomShawn) [reply tidb-lightning/deploy-tidb-lightning.md:200]: @kennytm Please help check whether the `status server` exactly means `TiKV Importer server`, thanks!

**Comment by Joyinqin** [comment]
@TomShawn PTAL

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 19]
```diff
@@ -12,14 +12,15 @@ TiDB Lightning 支持读取 CSV（逗号分隔值）的数据源，以及其他
 包含整张表的 CSV 文件需命名为 `db_name.table_name.csv`，该文件会被解析为数据库 `db_name` 里名为 `table_name` 的表。
 
 如果一个表分布于多个 CSV 文件，这些 CSV 文件命名需加上文件编号的后缀，如 `db_name.table_name.003.csv`。
+数字部分不需要连续，但必须递增并用零填充。
```
Comment (TomShawn) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:15]:
```suggestion
数字部分不需要连续但必须递增，并用零填充。
```

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 20]
```diff
@@ -136,6 +137,24 @@ TiDB Lightning 并不完全支持 `LOAD DATA` 语句中的所有配置项。例
 * 不可跳过表头（`IGNORE n LINES`）。如有表头，必须是有效的列名。
 * 定界符和分隔符只能为单个 ASCII 字符。
 
+## 严格格式
```
Comment (TomShawn) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:140]:
```suggestion
## 设置 `strict-format` 启用严格格式
```

File: tidb-lightning/migrate-from-csv-using-tidb-lightning.md [hunk 21]
```diff
@@ -136,6 +137,24 @@ TiDB Lightning 并不完全支持 `LOAD DATA` 语句中的所有配置项。例
...
+
+导入文件的大小统一约为 256 MB 时，TiDB Lightning 可达到最佳工作状态。如果导入单个 CSV 大文件，TiDB Lightning 只能使用一个线程来处理，这会降低导入速度。
+
+要解决此问题，可先将 CSV 文件分割为多个文件。对于通用 CSV 格式文件，在没有读取整个文件的情况下无法快速确定行的开始和结束位置。因此，默认情况下 TiDB Lightning 不会自动分割 CSV 文件。但如果你确定待导入的 CSV 文件符合特定的限制要求，则可以启用 `strict-format` 设置。启用后，TiDB Lightning 会将单个 CSV 大文件分割为多个 256 MB 的文件块进行并行处理。
```
Comment (TomShawn) [inline tidb-lightning/migrate-from-csv-using-tidb-lightning.md:144]:
```suggestion
要解决此问题，可先将 CSV 文件分割为多个文件。对于通用格式的 CSV 文件，在没有读取整个文件的情况下无法快速确定行的开始和结束位置。因此，默认情况下 TiDB Lightning 不会自动分割 CSV 文件。但如果你确定待导入的 CSV 文件符合特定的限制要求，则可以启用 `strict-format` 设置。启用后，TiDB Lightning 会将单个 CSV 大文件分割为单个大小为 256 MB 的多个文件块进行并行处理。
```

File: tidb-lightning/monitor-tidb-lightning.md [hunk 22]
```diff
@@ -43,16 +24,26 @@ pprof-port = 8289
...
 
-要让 Prometheus 发现 Lightning，可以将地址直接写入其配置文件，例如：
+监控的端口也可在 `tikv-importer.toml` 配置:
+
+```toml
+# 状态服务器的监听地址
+status-server-address = '0.0.0.0:8286'
+```
+
+需要配置 Prometheus 才能发现服务器。例如，将服务器地址直接添加至 `scrape_configs` 部分：
```
Comment (TomShawn) [inline tidb-lightning/monitor-tidb-lightning.md:34]:
```suggestion
配置 Prometheus 后，`tidb-lightning` 才能发现服务器。配置方法如下，将服务器地址直接添加至 `scrape_configs` 部分：
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 23]
```diff
@@ -59,6 +59,15 @@ table-concurrency = 6
 # 对于不同的存储介质，此参数可能需要调整以达到最佳效率。
 io-concurrency = 5
 
+[security]
+# 指定集群中用于 TLS 连接的证书和密钥。
+# CA 的公钥证书。如果留空，则可禁用 TLS。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:64]:
```suggestion
# CA 的公钥证书。如果留空，则禁用 TLS。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 24]
```diff
@@ -132,6 +141,18 @@ no-schema = false
 # 注意：**数据** 文件始终解析为 binary 文件。
 character-set = "auto"
 
+# 严格格式的导入数据可加快处理速度。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:144]:
```suggestion
# “严格”格式的导入数据可加快处理速度。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 25]
```diff
@@ -132,6 +141,18 @@ no-schema = false
...
+# strict-format = true 要求：
+# 在 CSV 文件的所有记录中，每条数据记录的值不可包含字符换行符（U+000A 和 U+000D，即 \r 和 \n）
+# 甚至使用转义符时都不可以，即严格将换行符作为行分隔符。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:147]:
```suggestion
# 甚至被引号包裹的字符换行符都不可包含，即换行符只可用来分隔行。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 26]
```diff
@@ -132,6 +141,18 @@ no-schema = false
...
+# 导入数据源为严格格式时，TiDB Lightning 会快速定位大文件的分割位置进行并行处理。
+# 但是如果输入数据为非严格格式，可能会将一条完整的数据分割成两部分，导致结果出错。
+# 为了保证数据安全而非追求速度，默认值为 false。
```
Comment (TomShawn) [inline tidb-lightning/tidb-lightning-configuration.md:150]:
```suggestion
# 为保证数据安全而非追求处理速度，默认值为 false。
```

**Pull request merged**